packages/sdk/src/cli/checkUnsupportedDataProxy.ts

```diff
@@ -68,15 +67,16 @@ async function _checkUnsupportedDataProxy(command: string, args: Args, implicitS
       const urlEnvVarValue = urlEnvVarName ? process.env[urlEnvVarName] : undefined
 
       if ((urlFromValue ?? urlEnvVarValue)?.startsWith('prisma://')) {
-        console.error(forbiddenCmdWithDataProxyFlagMessage(command))
-        process.exit(1)
+        return forbiddenCmdWithDataProxyFlagMessage(command)
       }
     }
   }
+
+  return undefined
 }
 
 export async function checkUnsupportedDataProxy(command: string, args: Args, implicitSchema: boolean) {
-  try {
-    await _checkUnsupportedDataProxy(command, args, implicitSchema)
-  } catch {}
+  const message = await checkUnsupportedDataProxyMessage(command, args, implicitSchema).catch(() => undefined)
```

Comment:
We get errors returns as strings and throw them as needed, removes `process.exit(1)`.